app/src/androidTest/java/org/mozilla/tv/firefox/ui/robots/NavigationOverlayRobot.kt

```diff
@@ -56,14 +59,18 @@ class NavigationOverlayRobot {
      * Navigate to the settings channel using keypresses
      */
     fun linearNavigateToSettings() {
-        // We hard-code this navigiation pattern because making a generic way to linearly navigate
-        // is very difficult within Espresso. Espresso supports asserting view state, but not
-        // querying it. Because of this, we can't write conditional logic based on the currently
-        // focused view.
+
+        var settingsTileSelected = false
+
         device.apply {
-            // This will need to change if the button layout changes. However, such layout
-            // changes are infrequent, and updating this will be easy.
-            repeat(4) { pressDPadDown() }
+            while (!settingsTileSelected) {
+                try {
+                    onView(allOf(withId(R.id.settings_cardview), hasFocus(true))).assertIsDisplayed(true)
+                    settingsTileSelected = true
```

Comment:
This looks good to me!

Though I'm a little confused why the number of clicks is different - doesn't our UI stay the same regardless of what device it's on?

Reply:
No, when the channel experiment is enabled, there are more rows between Pocket tile row and Settings menu row. since it gets enabled randomly, we have to dynamically adjust the number of DPADDOWN clicks.